Modules/Core/SpatialObjects/include/itkSpatialObjectToImageFilter.hxx

```diff
@@ -84,25 +84,18 @@ template <typename TInputSpatialObject, typename TOutputImage>
 void
 SpatialObjectToImageFilter<TInputSpatialObject, TOutputImage>::SetSpacing(const SpacingType & spacing)
 {
-  unsigned int i;
-
-  for (i = 0; i < TOutputImage::ImageDimension; ++i)
-  {
-    if (Math::NotExactlyEquals(static_cast<double>(spacing[i]), m_Spacing[i]))
-    {
-      break;
-    }
-  }
-  if (i < TOutputImage::ImageDimension)
+  if (ContainerCopyWithCheck(m_Spacing, spacing, TOutputImage::ImageDimension))
   {
-    for (i = 0; i < TOutputImage::ImageDimension; ++i)
+    this->Modified();
+    for (unsigned int i = 0; i < TOutputImage::ImageDimension; ++i)
     {
-      if (spacing[i] != 0)
+      if (spacing[i] <= 0)
       {
-        m_Spacing[i] = spacing[i];
+        itkExceptionMacro("Zero-valued spacing and negative spacings are not supported and may result in undefined "
```

Comment:
In the ImageBase class, the exception only occur with LEGACY_REMOVE. The conversion I have this having the DataObject block certain settings is that it may prevent users from loading and correcting existing data. 

It may be a better approach to negative spacing be a warning in the DataObject, but the ProcessObject's have an exception when the inputs have these invalid values. I don't know if the Spatial object have a base ProcessObject like the Image filters do. Just my 2 cents.

Reply:
itkSpatialObjectToImageFilter does derive from ProcessObject, so I think this implementation of throwing an exception here does conform to your recommendation.

This is a bug fix, and I don't know what the correct behavior should be without the Exception.

I appreciate your thoughts here.  I don't see how to easily implement a warning in the DataObject.